docs/changelog.md

```diff
@@ -1,6 +1,14 @@
 (changelog)=
 # Release notes
 
+## Verison 0.17.0
+*Unreleased*
+
+### Breaking changes
+
+- Removed iodide-specific code in `pyodide.js`. This breaks compatibility with
+  iodide.
```

Comment:
For future PRs, it's also helpful to include a link to the PR in the changelog entries. (I'll fix this one in https://github.com/iodide-project/pyodide/pull/987)